<a href="https://colab.research.google.com/github/HugoStigletz/Data-Analytics-Portfolio/blob/main/dj_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

You’re a budding DJ and your handle is DJ Data. You have
some cousins who have been DJing for a while and they’re
rolling their eyes at your humble beginnings. You have to
prove to them you have what it takes to be a DJ, and a good
one at that.
Part of what makes a good DJ is syncing with your audience and playing songs they like. You
have three gigs coming up that will be your DJ debut, so you want to make a good impression.
Using your smarts, you acquired a dataset that classifies songs into 14 different categories. You
know the audience type for each gig, so you want to create a song list you think each audience
type will like best.
Audience
After some thought, you assigned your audiences to these traits.
1. Loungers / quiet types
a. high acousticness
b. low energy
c. low loudness
2. Track team / partiers
a. high energy
b. high tempo
c. short duration
3. NYT bestseller book club / literary fans
a. high popularity
b. high danceability
c. high speechiness

Create artist list and visualize
Now that you have an idea what your audience is like, you can use your dataset to pick the
songs according to their classifications. We’re going to use Pandas library to interact with our
dataset, then use Plotly to plot a bar chart of each audience characteristic with the artist names
we need for our song list. Let’s create a Python notebook in Google Colab.


In [1]:
from google.colab import files
uploaded = files.upload()

Saving music_genre_classification.csv to music_genre_classification.csv


In [2]:
import pandas as pd
import plotly.express as px

LOUNGER / QUIET TYPES AUDIENCE

In [26]:
# Read CSV for the column you need for lounger audience
df1 = pd.read_csv("music_genre_classification.csv", usecols
= ['Artist Name', 'energy', 'loudness', 'acousticness'])
# Get the smallest 10 energy values
low_energy = df1.nsmallest(10, 'energy')
print(low_energy[['Artist Name', 'energy']])
# Plot a bar chart, using plotly
px.bar(low_energy, 'Artist Name', 'energy', title='10 Lowest Energy Songs by Artist',
color='energy', color_continuous_scale='Greens')

                                            Artist Name   energy
6089                               Todd Anderson-Kunert  0.00002
2801         Ludwig van Beethoven, Emanuel Ax, Yo-Yo Ma  0.00433
1521  G. I. Gurdjieff, Anja Lechner, Vassilis Tsabro...  0.00526
7432                                              HILMA  0.00700
3561  Luigi Boccherini, Tim Hugh, Scottish Chamber O...  0.00714
1985                              Gerry Mulligan Sextet  0.00894
2919                                 Broccoli, you too?  0.00976
3373             Francisco Villegas, Julio Mariñelarena  0.01040
5833                                      Library Tapes  0.01080
3997                                        Carlos Cipa  0.01230


In [27]:
# Get the lowest 10 loudness values in decibels
quietest = df1.nsmallest(10, 'loudness')
print(quietest[['Artist Name', 'loudness']])

# Plot to a bar chart in plotly
px.bar(quietest, 'Artist Name', 'loudness', title= "10 Quietest Songs by Artist", color='loudness', color_continuous_scale='blues')

                                            Artist Name  loudness
3561  Luigi Boccherini, Tim Hugh, Scottish Chamber O...   -36.544
2801         Ludwig van Beethoven, Emanuel Ax, Yo-Yo Ma   -32.639
1822  Alexander Glazunov, Alexander Rudin, Moscow Ph...   -32.520
3150                      Victor Kozlov, Irina Kulikova   -31.846
1268  Johann Sebastian Bach, Freiburger Barockorchester   -30.625
7432                                              HILMA   -29.955
1521  G. I. Gurdjieff, Anja Lechner, Vassilis Tsabro...   -29.845
6651              Ottorino Respighi, Magdalena Hoffmann   -29.234
3997                                        Carlos Cipa   -29.161
4895                                      Tony Anderson   -28.958


In [36]:
# Get the highest 10 acousticness values
best_acoustics = df1.nlargest(30, 'acousticness')
print(best_acoustics[['Artist Name', 'acousticness']])

# Plot to a bar chart in plotly
px.bar(best_acoustics, 'Artist Name', 'acousticness', title
='10 Best Acoustic Songs by Artist', color='acousticness',
color_continuous_scale='sunsetdark')

                                      Artist Name  acousticness
757                   Memphis Minnie & Kansas Joe         0.995
1264                             Blind Boy Fuller         0.995
3206                              Baby Boy Warren         0.995
5244             Anja Lechner, François Couturier         0.995
6349                                Raquel Meller         0.995
1404                             Daniel Raticelli         0.994
2124                                Juanita Reina         0.994
3649                              Thelonious Monk         0.994
4237                         Antonio Garcia Isaac         0.994
6154                                   R.J. Smith         0.994
6426  Franz Schubert, Mischa Maisky, Daria Hovora         0.994
6700                                 Diane Edison         0.994
630                            Juanito Valderrama         0.993
688                               Lata Mangeshkar         0.993
893             Rekha, Vinod Khanna, Ram

Run that cell and see your small table print and bar chart plot. What do you notice about
this bar chart that’s different from the others? Why do you think it’s different?

Try changing the 10 value in the nlargest() function to 30, then run the cell
again. Is it more clear why the bar chart was different?

The number of values that are placed in your input as well as the value of that input could effect the outcome of your visual

--------------------------------------------------------------------------------

**TRACK TEAM / PARTIERS**

In [7]:
# Read CSV for the column you need for track audience
df2 = pd.read_csv("music_genre_classification.csv", usecols
= ['Artist Name', 'energy', 'tempo', 'duration_in min/ms'])
# Get the smallest 10 energy values
high_energy = df2.nlargest(10, 'energy')
print(high_energy[['Artist Name', 'energy']])
# Plot a bar chart, using plotly
px.bar(high_energy, 'Artist Name', 'energy', title='10 Highest Energy Songs by Artist',
color='energy', color_continuous_scale='Mint')

            Artist Name  energy
370          Darkthrone   1.000
15            FEVER 333   0.998
4241          FEVER 333   0.998
6208               Lard   0.998
1387     Pretty Vicious   0.997
1650         Hellripper   0.997
4934         Hellhammer   0.997
5197  The Acacia Strain   0.997
283       Ozzy Osbourne   0.996
959        Funeral Mist   0.996


In [8]:
# Get the highest 10 tempo values in decibels
high_tempo = df2.nlargest(10, 'tempo')
print(high_tempo[['Artist Name', 'tempo']])

# Plot to a bar chart in plotly
px.bar(high_tempo, 'Artist Name', 'tempo', title= "10 Higest_Tempo Songs by Artist", color='tempo', color_continuous_scale='ice')

            Artist Name    tempo
3581                IDK  249.438
2579         New Candys  216.194
5150         New Candys  216.194
4424         frown line  216.091
7018  Angels & Airwaves  216.020
5465    Ólafur Arnalds  212.849
4385           AFN Peso  212.166
2390               Evil  209.657
94          Aisha Badru  208.648
3688      Johnny Winter  208.576


In [9]:
# Get the shortest 10 duration_in min/ms values
shortest_duration = df2.nsmallest(10, 'duration_in min/ms')
print(shortest_duration[['Artist Name', 'duration_in min/ms']])

# Plot to a bar chart in plotly
px.bar(shortest_duration, 'Artist Name', 'duration_in min/ms', title
='10 Shortest_Duration Songs by Artist', color='duration_in min/ms',
color_continuous_scale='delta')

                                Artist Name  duration_in min/ms
4010                        Mahendra Kapoor            0.687450
2132                           R. D. Burman            1.209367
4066             Ilaiyaraaja, K. J. Yesudas            1.291667
5974   Eleni Karaindrou, Alexandros Botinis            1.353333
6447                   Kavita Krishnamurthy            1.426883
725                           Reyanna Maria            1.429033
4274                           R. D. Burman            1.434450
609                            Bappi Lahiri            1.464217
5438  Mani Sharma, Amit Heri, Rajesh Vaidya            1.495367
7561                               Labrinth            1.531167


--------------------------------------------------------------------------------

NYT BESTSELLER BOOK CLUB / LITERARY FANS

In [15]:
# Read CSV for the column you need for NYT audience
df3 = pd.read_csv("music_genre_classification.csv", usecols
= ['Artist Name', 'Popularity', 'danceability', 'speechiness'])
# Get the Top 10 Popularity values
high_popularity = df3.nlargest(10, 'Popularity')
print(high_popularity[['Artist Name', 'Popularity']])
# Plot a bar chart, using plotly
px.bar(high_popularity, 'Artist Name', 'Popularity', title='10 Highest Popularity Songs by Artist',
color='Popularity', color_continuous_scale='twilight')

                  Artist Name  Popularity
6062                Lil Nas X       100.0
3797            Justin Bieber        99.0
2359                 Doja Cat        98.0
1816              Masked Wolf        97.0
6736               Ed Sheeran        97.0
3315               Bruno Mars        96.0
4263                Bad Bunny        96.0
6038  J Balvin, Maria Becerra        95.0
777                The Weeknd        94.0
4190            Billie Eilish        93.0


In [17]:
# Get the highest 10 Best_Danceability values in decibels
Best_Danceability = df3.nlargest(10, 'danceability')
print(Best_Danceability[['Artist Name', 'danceability']])

# Plot to a bar chart in plotly
px.bar(Best_Danceability, 'Artist Name', 'danceability', title= "10 Best_Danceability Songs by Artist", color='danceability', color_continuous_scale='ice')

                   Artist Name  danceability
2326                Rebelution         0.975
4120                Gucci Mane         0.972
7116                      Tyga         0.968
3868               Jody Breeze         0.966
4930              Run–D.M.C.         0.965
4019  Ashnikko, Princess Nokia         0.964
4215                  Ashnikko         0.964
1576                 Boy 8-Bit         0.962
5417                Lou Hayter         0.962
6070                    John K         0.958


In [44]:
# Get the highest 10 speechiness values
highest_speechiness = df3.nlargest(10, 'speechiness')
print(highest_speechiness[['Artist Name', 'speechiness']])

# Plot to a bar chart in plotly
px.bar(highest_speechiness, 'Artist Name', 'speechiness', title
='10 Highest_Speechiness Songs by Artist', color='speechiness',
color_continuous_scale='speed')

                                            Artist Name  speechiness
1043                                       Ishmael Reed        0.960
5763                                       Amiri Baraka        0.938
7005  Rajesh Khanna, Sumita Sanyal, Ramesh Deo, Seem...        0.829
734                                    Amitabh Bachchan        0.814
4740                                    Langston Hughes        0.806
580                                     Margaret Walker        0.678
1602                                           Rob Frye        0.667
3646                                          Son House        0.658
4446                                       Moneybagg Yo        0.619
893                   Rekha, Vinod Khanna, Ram P. Sethi        0.565


Look closely at the track team charts for energy and tempo. There’s a duplicate in each.

How does that interrupt your bar chart?

What should have been done with the data before plotting it to avoid this?
(Check your data for duplicate information)




After you complete the bar charts for all three audience types, look for overlap. 

 Write those down as must-have songs for your song list.

2801         Ludwig van Beethoven, Emanuel Ax, Yo-Yo Ma  

1521  G. I. Gurdjieff, Anja Lechner, Vassilis Tsabropoulos  

7432                                              HILMA  

3561  Luigi Boccherini, Tim Hugh, Scottish Chamber Orchestra, Anthony Halstead        

3997                                        Carlos Cipa 